shopinvader_api_address/routers/address_service.py

```diff
@@ -0,0 +1,159 @@
+    BillingAddress,
+    ShippingAddress,
+)
+
+from ..schemas import AddressInput, AddressUpdate
+
+# create a router
+address_router = APIRouter(tags=["addresses"])
+
+#### Billing addresses ####
+
+
+@address_router.get("/addresses/billing", response_model=BillingAddress)
+def get_billing_address(
+    env: Environment = Depends(authenticated_partner_env),
+    partner=Depends(authenticated_partner),
+) -> BillingAddress:
+    """
+    Get billing address of authenticated user
+    billing address corresponds to authenticated partner
+    """
+    address_id = partner._get_shopinvader_billing_addresses()
+    return BillingAddress.from_res_partner(address_id)
+
+
+@address_router.post(
+    "/addresses/billing", response_model=BillingAddress, status_code=201
+)
+def update_billing_address(
+    data: AddressUpdate,
+    env: Environment = Depends(authenticated_partner_env),
+    partner=Depends(authenticated_partner),
+) -> BillingAddress:
+    """
+    Update billing address of authenticated user
+    billing address corresponds to authenticated partner
+    """
+    vals = env["shopinvader_api_address.mapper"]._addressupdate_to_res_partner(data)
+    address_id = partner._update_shopinvader_billing_addresses(vals)
+    return BillingAddress.from_res_partner(address_id)
+
+
+#### Shipping address ####
+
+
+@address_router.get(
+    "/addresses/shipping", response_model=PagedCollection[ShippingAddress]
+)
+@address_router.get(
+    "/addresses/shipping/{id}", response_model=PagedCollection[ShippingAddress]
+)
+def get_shipping_address(
+    env: Environment = Depends(authenticated_partner_env),
+    partner=Depends(authenticated_partner),
+    id: int | None = None,
+) -> PagedCollection[ShippingAddress]:
+    """
+    Get shipping addresses of authenticated user
+    Can be used to get every shipping address: /addresses/shipping
+    Can be used to get one specific address: /addresses/shipping/rec_id
+    """
+    address_ids = partner._get_shopinvader_shipping_addresses(id)
+    return PagedCollection[ShippingAddress](
+        total=len(address_ids),
+        items=[ShippingAddress.from_res_partner(rec) for rec in address_ids],
+    )
+
+
+@address_router.post(
+    "/addresses/shipping", response_model=ShippingAddress, status_code=201
+)
+@address_router.post("/addresses/shipping/{id}", response_model=ShippingAddress)
+def create_update_shipping_address(
+    data: AddressInput,
+    env: Environment = Depends(authenticated_partner_env),
+    partner=Depends(authenticated_partner),
+    id: int | None = None,
+) -> ShippingAddress:
+    """
+    Create/Update shipping address of authenticated user
+    """
+    vals = env["shopinvader_api_address.mapper"]._addressInput_to_res_partner(data)
+    if id is None:
+        address_id = partner._create_shopinvader_shipping_addresses(
+            vals
+        )
+    else:
+        address_id = partner._update_shopinvader_shipping_addresses(vals, id)
```

Comment:
```suggestion
        address = partner._update_shopinvader_shipping_addresses(vals, id)
```

This is not an id, so `address` reads better.
Use the _id suffix for variables that are really id (and ORM field names because it's the Odoo convention we can't change it)